src/lib/x509/x509path.h

```diff
@@ -330,13 +349,35 @@ Path_Validation_Result BOTAN_PUBLIC_API(2, 0)
 */
 namespace PKIX {
 
-Certificate_Status_Code build_all_certificate_paths(std::vector<std::vector<X509_Certificate>>& cert_paths,
-                                                    const std::vector<Certificate_Store*>& trusted_certstores,
-                                                    const std::optional<X509_Certificate>& end_entity,
-                                                    const std::vector<X509_Certificate>& end_entity_extra);
+/**
+* Create all certificate paths by identifying all possible routes from the
+* end-entity certificate to any certificate in the certificate store list. Paths
+* may also end in intermediate or leaf certificates found in the certificate
+* stores.
+*
+* WARNING: The validity (e.g. signatures or constraints) of the output path IS
+* NOT checked.
+*
+* @param cert_paths output parameter to be filled with all discovered certificate paths
+* @param trusted_certstores list of certificate stores that contain trusted certificates
+* @param end_entity the cert to be validated
+* @param end_entity_extra optional list of additional untrusted certs for path building
+* @return result of the path building operation (OK or error)
+*/
+Certificate_Status_Code BOTAN_PUBLIC_API(3, 10)
+   build_all_certificate_paths(std::vector<std::vector<X509_Certificate>>& cert_paths,
+                               const std::vector<Certificate_Store*>& trusted_certstores,
+                               const X509_Certificate& end_entity,
+                               const std::vector<X509_Certificate>& end_entity_extra);
 
 /**
-* Build certificate path
+* Same as build_all_certificate_paths but only outputs a single path. If there are
+* paths ending in self-signed certificates, these are prioritized  over paths ending
+* in intermediate or leaf certificates of the certificate store.
+*
+* WARNING: The validity (e.g. signatures or constraints) of the output path IS
+* NOT checked.
+*
```

Comment:
```suggestion
* Same as build_all_certificate_paths but only outputs a single path. If there are
* paths ending in self-signed certificates, these are prioritized over paths ending
* in intermediate or leaf certificates of the certificate store.
*
* WARNING: The validity (e.g. signatures or constraints) of the output path IS
* NOT checked.
*
```

... sometimes I'm a kerning nerd. ;)